scripts/evaluate.py

```diff
@@ -1149,6 +1149,69 @@ def validator(inputs):
             return [np_result,]
         return (mod, {}, {"data": input_shape}, dtype, target, validator)
+    
+
+    def export_lib_and_meta(self, model_name, target="llvm", dtype="float32", output_dir="./"):
+        name_to_model = {
+        "avg_pooling": self.import_avg_pooling,
+        "concat": self.import_concat,
+        "conv2d": self.import_conv2d,
+        "conv2d_3x3": self.import_conv2d_3x3,
+        "conv2d_a6x_compiler_hang": self.import_conv2d_a6x_compiler_hang,
+        "conv2d_a6x_compiler_hang_kernel_single_op": self.import_conv2d_a6x_compiler_hang_kernel_single_op,
+        "conv2d_conv2d": self.import_conv2d_conv2d,
+        "conv2d_deeplab_1_nchwc": self.import_conv2d_deeplab_1_nchwc,
+        "conv2d_inceptionv3_nchw_3c": self.import_conv2d_inceptionv3_nchw_3c,
+        "conv2d_mem_reuse": self.import_conv2d_mem_reuse,
+        "conv2d_mem_reuse1": self.import_conv2d_mem_reuse1,
+        "conv2d_nchw": self.import_conv2d_nchw,
+        "conv2d_nhwc": self.import_conv2d_nhwc,
+        "conv2d_resnet50_v2_nchw_3c": self.import_conv2d_resnet50_v2_nchw_3c,
+        "conv2d_x4": self.import_conv2d_x4,
+        "conv2d_yolov3_v2_nchw_3c": self.import_conv2d_yolov3_v2_nchw_3c,
+        "deeplabv3": self.import_deeplabv3,
+        "depthwise_conv2d": self.import_depthwise_conv2d,
+        "depthwise_conv2d_nchwc": self.import_depthwise_conv2d_nchwc,
+        "depthwise_conv2d_nhwc": self.import_depthwise_conv2d_nhwc,
+        "global_pooling": self.import_global_pooling,
+        "inceptionv3": self.import_inceptionv3,
+        "layout_transform_contract": self.import_layout_transform_contract,
+        "layout_transform_expand": self.import_layout_transform_expand,
+        "mace_deeplabv3": self.import_mace_deeplabv3,
+        "mace_inceptionv3": self.import_mace_inceptionv3,
+        "mace_mobilenetv1_nchw": self.import_mace_mobilenetv1_nchw,
+        "mace_mobilenetv1_nhwc": self.import_mace_mobilenetv1_nhwc,
+        "mace_resnet50_v2": self.import_mace_resnet50_v2,
+        "mace_yolov3": self.import_mace_yolov3,
+        "max_pooling": self.import_max_pooling,
+        "max_pooling2": self.import_max_pooling2,
+        "mobilenetv1": self.import_mobilenetv1,
+        "mobilenetv3_ssdlite": self.import_mobilenetv3_ssdlite,
+        "resnet50": self.import_resnet50,
+        "resnet50_v2": self.import_resnet50_v2,
+        "vgg16": self.import_vgg16,
+        "yolov3_mxnet": self.import_yolov3_mxnet
+        }
+
+        import_method = name_to_model[model_name]
+        (mod, params, shape_dict, dtype, target, validator) = import_method(target, dtype)
+
+        with relay.build_config(opt_level=3):
+            graph, lib, params = relay.build(
+                mod, target_host=target, target=target, params=params
+            )
+
+        if args.output != './':
+            output_dir = output_dir + "/"
```

Comment:
This is not a universal solution. For example, on Windows, the path separator looks like this: `\`.
Delete this `if-else` block and use the `os.path.join` below:
```Python
lib.export_library(os.path.join(output_dir, model_name + "-default.so"))
```

Reply:
One more thing args.output is "./" by default which is Linux format. Nevertheless please use os.path or pathlib libraries for path processing. In the latter case you do not need to follow the path concatination and condition can be skipped:
`lib.export_library(os.path.join(args.output, model_name + "-default.so"))`